Gems/OpenXRVk/Code/Source/Builders/OpenXRVkAssetsBuilder.cpp

```diff
@@ -0,0 +1,310 @@
+        {
+            AssetBuilderSDK::JobDescriptor jobDescriptor;
+            // Very high priority because this asset is required to initialize the OpenXR runtime
+            // and initialize the I/O actions system.
+            jobDescriptor.m_priority = 1000;
+            jobDescriptor.m_critical = true;
+            jobDescriptor.m_jobKey = InteractionProfilesAssetJobKey;
+            jobDescriptor.SetPlatformIdentifier(platformInfo.m_identifier.c_str());
+            response.m_createJobOutputs.emplace_back(AZStd::move(jobDescriptor));
+        } // for all request.m_enabledPlatforms
+
+        response.m_result = AssetBuilderSDK::CreateJobsResultCode::Success;
+    }
+
+    
+    void OpenXRAssetsBuilder::ProcessInteractionProfilesAssetJob([[maybe_unused]] const AssetBuilderSDK::ProcessJobRequest& request, [[maybe_unused]] AssetBuilderSDK::ProcessJobResponse& response) const
+    {
+        // Open the file, and make sure there's no redundant data, the OpenXR Paths are well formatted, etc.
+       auto interactionProfilesAssetPtr = LoadAssetAsUniquePtr<OpenXRVk::OpenXRInteractionProfilesAsset>(request.m_fullPath);
+       if (!interactionProfilesAssetPtr)
+       {
+           AZ_Error(LogName, false, "Failed to load interaction profile source asset [%s]", request.m_fullPath.c_str());
+           response.m_resultCode = AssetBuilderSDK::ProcessJobResult_Failed;
+           return;
+       }
+
+       auto outcome = OpenXRVkAssetsValidator::ValidateInteractionProfilesAsset(*interactionProfilesAssetPtr.get());
+       if (!outcome.IsSuccess())
+       {
+           AZ_Error(LogName, false, "Invalid InteractionProfilesAsset [%s]. Reason:\n%s",
+               request.m_fullPath.c_str(), outcome.GetError().c_str());
+           response.m_resultCode = AssetBuilderSDK::ProcessJobResult_Failed;
+           return;
+       }
+
+       // We keep exact same asset name and extension.
+       AZStd::string assetFileName;
+       AZ::StringFunc::Path::GetFullFileName(request.m_fullPath.c_str(), assetFileName);
+
+       // Construct product full path
+       AZStd::string assetOutputPath;
+       AzFramework::StringFunc::Path::ConstructFull(request.m_tempDirPath.c_str(), assetFileName.c_str(), assetOutputPath, true);
+
+       bool result = AZ::Utils::SaveObjectToFile(assetOutputPath, AZ::DataStream::ST_XML, interactionProfilesAssetPtr.get());
+       if (result == false)
+       {
+           AZ_Error(LogName, false, "Failed to save asset to %s", assetOutputPath.c_str());
+           response.m_resultCode = AssetBuilderSDK::ProcessJobResult_Failed;
+           return;
+       }
+
+       AssetBuilderSDK::JobProduct jobProduct;
+       if (!AssetBuilderSDK::OutputObject(interactionProfilesAssetPtr.get(), assetOutputPath,
+           azrtti_typeid<OpenXRVk::OpenXRInteractionProfilesAsset>(),
+           aznumeric_cast<uint32_t>(0), jobProduct))
+       {
+           AZ_Error(LogName, false, "FIXME this message.");
```

Comment:
Is FIXME a placeholder? Can you improve the error message?

Reply:
Fixed.